packages/edit-navigation/src/components/menu-editor/navigation-structure-panel.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	__experimentalBlockNavigationToolbar as BlockNavigationToolbar,
+	__experimentalBlockNavigationList as BlockNavigationList,
+} from '@wordpress/block-editor';
+import { Panel, PanelBody } from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+
+export default function NavigationStructurePanel( { blocks } ) {
+	const selectedBlockClientIds = useSelect( ( select ) =>
+		select( 'core/block-editor' ).getSelectedBlockClientIds()
+	);
```

Comment:
Same optimization here:
```suggestion
	const selectedBlockClientIds = useSelect( ( select ) => {
		select( 'core/block-editor' ).getSelectedBlockClientIds()
	}, [] );
```